gnd/src/main/java/com/google/android/gnd/ui/observationdetails/ObservationDetailsFragment.java

```diff
@@ -156,35 +149,25 @@ private void addField(Field field, Observation observation) {
     observation
         .getResponses()
         .getResponse(field.getId())
-        .map(r -> r.getDetailsText(field))
         .ifPresent(
-            value -> {
-              if (field.getType().equals(Type.PHOTO)) {
-                binding.fieldValue.setVisibility(View.GONE);
-                binding.imagePreview.setVisibility(View.VISIBLE);
-
-                // TODO: Subscriptions should only be made inside lifecycle methods so they can be
-                // properly disposed of in their equivalent end lifecycle methods. To do this
-                // safely, we can create a PublishSubject in the view model that gets exposed to
-                // the fragment as a LiveData.
-                storageManager
-                    .getDownloadUrl(value)
-                    .subscribeOn(schedulers.io())
-                    .observeOn(schedulers.ui())
-                    .as(autoDisposable(this))
-                    .subscribe(
-                        uri ->
-                            Picasso.get()
-                                .load(uri)
-                                .placeholder(R.drawable.ic_photo_grey_600_24dp)
-                                .into(binding.imagePreview));
-
+            response -> {
+              if (field.getType() == Type.PHOTO) {
+                addPhotoField((ViewGroup) binding.getRoot(), field, response);
               } else {
-                binding.fieldValue.setText(value);
+                binding.fieldValue.setText(response.getDetailsText(field));
               }
             });
   }
 
+  private void addPhotoField(ViewGroup container, Field field, Response response) {
+    PhotoFieldBinding photoFieldBinding = PhotoFieldBinding.inflate(getLayoutInflater());
+    PhotoFieldViewModel photoFieldViewModel = viewModelFactory.create(PhotoFieldViewModel.class);
+    photoFieldBinding.setLifecycleOwner(this);
+    photoFieldBinding.setViewModel(photoFieldViewModel);
+    photoFieldViewModel.updateField(response, field);
```

Comment:
It appears this method is called both from `init()` and here. Is this intentional? If so, why?

Reply:
`init()` takes an `ObservableMap` as a parameter and calls `updateField()` whenever the response is changed. Here, we don't call `init()` as this is a one time update.

Reply:
Makes me wonder if `init()` should a part of `PhotoFieldViewModel` at all....